runtime/src/accounts_db.rs

```diff
@@ -611,21 +579,37 @@ impl AccountsDB {
             } else {
                 let mut no_delete = false;
                 for (_slot, account_info) in account_infos {
-                    if *store_counts.get(&account_info.store_id).unwrap() != 0 {
+                    if store_counts.get(&account_info.store_id).unwrap().0 != 0 {
                         no_delete = true;
                         break;
                     }
                 }
                 no_delete
             };
             if no_delete {
+                let mut pending_store_ids: HashSet<usize> = HashSet::new();
                 for (_slot_id, account_info) in account_infos {
-                    Self::inc_store_counts(
-                        account_info.store_id,
-                        &purges,
-                        store_counts,
-                        &mut already_counted,
-                    );
+                    if !already_counted.contains(&account_info.store_id) {
```

Comment:
Is this `already_counted` still necessary? Seems like all stores in `pending_store_ids` will be unique/only processed once

Reply:
I think it's needed. Even with the case where you have two accounts depend on the same slot, they will enqueue each other forever if it's not checked.

Reply:
nvm, i see it's necessary for line 608